### DISCUSSION TITLE
Clarification on SendPaymentV2 and dest_features

### DISCUSSION THREAD

**Discussion by gtochim** [lightningnetwork/lnd#8847] [General]
When hitting [SendPaymentV2](https://api.lightning.community/api/lnd/router/send-payment-v2/index.html) and providing an invoice in the payment_request field, the destination feature vector is decoded in [zpay32/decode.go](https://github.com/lightningnetwork/lnd/blob/master/zpay32/decode.go) . No validation is done. However, if one doesn't provide an invoice and provides among other necessary fields, a dest_features , this vector is parsed in the [router_backend](https://github.com/lightningnetwork/lnd/blob/master/lnrpc/routerrpc/router_backend.go) via UnmarshalFeatures, which does some validation, e.g. checking that both OPT and REQ bits are not set at the same time.

Is this difference in behavior intended? For what reason?

**Comment by ellemouton** [comment]
@gtochim  - im assuming that this was just an oversight :)
